cylc/flow/task_job_mgr.py

```diff
@@ -273,10 +273,10 @@ def submit_task_jobs(self, workflow, itasks, curve_auth,
             for itask in itasks:
                 # If there are any hosts left for this platform which we
                 # have not previously failed to contact with a 255 error.
-                if (
-                    set(itask.platform['hosts']) -
-                    self.task_remote_mgr.bad_hosts
-                ):
+                if [
+                    host for host in list(set(itask.platform['hosts']))
+                    if host not in self.task_remote_mgr.bad_hosts
+                ]:
```

Comment:
As above

Reply:
Ay, one more instance and it becomes a helper function.